### ISSUE TITLE
8.2: Failed to build ghcjs-prim while booting

### ISSUE THREAD

**Issue by ElvishJerricco** [ghcjs/ghcjs#637]
Building the 8.2 branch, I get this error when booting gets to ghcjs-prim:

```
Building library for ghcjs-prim-0.1.1.0..
[1 of 3] Compiling GHCJS.Prim       ( GHCJS/Prim.hs, dist/build/GHCJS/Prim.js_o )

GHCJS/Prim.hs:96:15: error:
    * Variable not in scope: unpackCString# :: t -> String
    * Perhaps you meant one of these:
        `GHC.unpackCString#' (imported from GHC.CString),
        `unpackJSString#' (line 206),
        `GHC.unpackCStringUtf8#' (imported from GHC.CString)
   |
96 |   toJSString (unpackCString# a) = JSVal (unsafeUnpackJSString## a)
   |               ^^^^^^^^^^^^^^

GHCJS/Prim.hs:96:42: error:
    * Variable not in scope: unsafeUnpackJSString## :: t -> ByteArray#
    * Perhaps you meant one of these:
        `unsafeUnpackJSString#' (line 221),
        `unsafeUnpackJSStringUtf8#' (line 227),
        `unsafeUnpackJSStringUtf8##' (line 231)
   |
96 |   toJSString (unpackCString# a) = JSVal (unsafeUnpackJSString## a)
   |                                          ^^^^^^^^^^^^^^^^^^^^^^

GHCJS/Prim.hs:98:15: error:
    * Variable not in scope: unpackCStringUtf8# :: Addr# -> String
    * Perhaps you meant one of these:
        `GHC.unpackCStringUtf8#' (imported from GHC.CString),
        `unpackJSStringUtf8#' (line 211), `unpackJSStringUtf8##' (line 216)
   |
98 |   toJSString (unpackCStringUtf8# a) = JSVal (unsafeUnpackJSStringUtf8## a)
   |               ^^^^^^^^^^^^^^^^^^
Failed to install ghcjs-prim-0.1.1.0
cabal: Error: some packages failed to install:
ghcjs-prim-0.1.1.0-Lw1qzAJ9IR31NFzWC6t27Z failed during the building phase.
The exception was:
ExitFailure 1

ghcjs-boot: error running: /nix/store/kj9ckr15a98vmmms0p0dafwvqp5m431s-cabal-install-2.2.0.0/bin/cabal --config-file /nix/store/iqv16j2p942xwf6cmkr2wlb04b7f88gm-ghcjs-8.2.0.1/lib/ghcjs-8.2.0.1/cabalBootConfig --ignore-sandbox install ./pkg/ghcjs-prim --allow-boot-library-installs --global --ghcjs --one-shot --avoid-reinstalls --builddir dist --with-compiler /nix/store/iqv16j2p942xwf6cmkr2wlb04b7f88gm-ghcjs-8.2.0.1/bin/ghcjs --with-hc-pkg /nix/store/iqv16j2p942xwf6cmkr2wlb04b7f88gm-ghcjs-8.2.0.1/bin/ghcjs-pkg --prefix /nix/store/iqv16j2p942xwf6cmkr2wlb04b7f88gm-ghcjs-8.2.0.1/lib/ghcjs-8.2.0.1 --enable-documentation --haddock-html --haddock-hoogle --haddock-hyperlink-source --enable-debug-info --disable-library-stripping --disable-executable-stripping -fghci --enable-profiling --ghcjs-options=-j4
exit status: 1
stderr: Warning: the --global flag is deprecated -- it is generally considered a bad
idea to install packages into the global store
Warning: No remote package servers have been specified. Usually you would have
one specified in the config file.

GHCJS/Prim.hs:96:15: error:
    * Variable not in scope: unpackCString# :: t -> String
    * Perhaps you meant one of these:
        `GHC.unpackCString#' (imported from GHC.CString),
        `unpackJSString#' (line 206),
        `GHC.unpackCStringUtf8#' (imported from GHC.CString)
   |
96 |   toJSString (unpackCString# a) = JSVal (unsafeUnpackJSString## a)
   |               ^^^^^^^^^^^^^^

GHCJS/Prim.hs:96:42: error:
    * Variable not in scope: unsafeUnpackJSString## :: t -> ByteArray#
    * Perhaps you meant one of these:
        `unsafeUnpackJSString#' (line 221),
        `unsafeUnpackJSStringUtf8#' (line 227),
        `unsafeUnpackJSStringUtf8##' (line 231)
   |
96 |   toJSString (unpackCString# a) = JSVal (unsafeUnpackJSString## a)
   |                                          ^^^^^^^^^^^^^^^^^^^^^^

GHCJS/Prim.hs:98:15: error:
    * Variable not in scope: unpackCStringUtf8# :: Addr# -> String
    * Perhaps you meant one of these:
        `GHC.unpackCStringUtf8#' (imported from GHC.CString),
        `unpackJSStringUtf8#' (line 211), `unpackJSStringUtf8##' (line 216)
   |
98 |   toJSString (unpackCStringUtf8# a) = JSVal (unsafeUnpackJSStringUtf8## a)
   |               ^^^^^^^^^^^^^^^^^^
```

**Comment by luite** [comment]
Looks like I messed up some merge

**Comment by luite** [comment]
sorry, I've been stuck debugging an unrelated problem with reflex on 8.2, I'll fix this soon.

**Comment by luite** [comment]
fixed now (and the other problem too, was #646)
